dashboard/app/controllers/home_controller.rb

```diff
@@ -91,10 +91,12 @@ def teacher_announcements
   private
 
   def should_redirect_to_script_overview?
-    if current_user.user_script_with_most_recent_progress
-      return current_user.most_recent_progress_in_recently_assigned_script? || current_user.last_assignment_after_most_recent_progress?
-    else
-      return true
+    if current_user.student? && !account_takeover_in_progress? && current_user.most_recently_assigned_user_script
+      if current_user.user_script_with_most_recent_progress
+        return current_user.most_recent_progress_in_recently_assigned_script? || current_user.last_assignment_after_most_recent_progress?
+      else
+        return true
+      end
```

Comment:
If your first if-expression is false this method returns `nil` instead of `true` or `false`, which doesn't break behavior but is a little messy.

This method could also be expressed as a single expression.  I think this is equivalent:
```rb
def should_redirect_to_script_overview?
  current_user.student? &&
  !account_takeover_in_progress &&
  current_user.most_recently_assigned_user_script &&
  (
    !current_user.user_script_with_most_recent_progress || 
    current_user.most_recent_progress_in_recently_assigned_script? || 
    current_user.last_assignment_after_most_recent_progress?
  )
end
```